doppler/resource_config.go

```diff
@@ -36,23 +39,89 @@ func resourceConfig() *schema.Resource {
 				Type:        schema.TypeString,
 				Required:    true,
 			},
+			"descriptor": {
+				Description: "The descriptor (project.config) of the Doppler config",
+				Type:        schema.TypeString,
+				Required:    false,
+				Computed:    true,
+			},
+			"inheritable": {
+				Description: "Whether or not the Doppler config can be inherited by other configs",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"inherits": {
+				Description: "A list of other Doppler config descriptors that this config inherits from. Descriptors match the format \"project.config\" (e.g. backend.stg), which is most easily retrieved as the computed descriptor of a doppler_config resource (e.g. doppler_config.backend_stg.descriptor)",
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
 		},
 	}
 }
 
+func inheritsArgToDescriptors(inherits []interface{}) ([]ConfigDescriptor, error) {
+	var descriptors []ConfigDescriptor
+
+	for _, descriptor := range inherits {
+		split := strings.Split(descriptor.(string), ".")
+		if len(split) != 2 {
+			return nil, fmt.Errorf("Unable to parse [%s] as descriptor", descriptor)
+		}
+		descriptors = append(descriptors, ConfigDescriptor{Project: split[0], Config: split[1]})
+	}
+
+	return descriptors, nil
+}
+
 func resourceConfigCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	client := m.(APIClient)
 
 	var diags diag.Diagnostics
 	project := d.Get("project").(string)
 	environment := d.Get("environment").(string)
 	name := d.Get("name").(string)
+	inheritable := d.Get("inheritable").(bool)
+	inherits := d.Get("inherits").([]interface{})
+
+	var config *Config
+	var err error
+
+	if name == environment {
+		// By definition, root configs share the same name as their environment. If the user attempted to define
+		// a resource for the root config (which would have required an environment to already be created), we
+		// should just fetch the root config instead of attempting to create it, which would fail.
+		config, err = client.GetConfig(ctx, project, name)
+	} else {
+		config, err = client.CreateConfig(ctx, project, environment, name)
+	}
 
-	config, err := client.CreateConfig(ctx, project, environment, name)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
+	if err = d.Set("descriptor", fmt.Sprintf("%s.%s", config.Project, config.Name)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if config.Inheritable != inheritable {
+		// Configs are always created as not inheritable, and inheritability cannot be specified during the creation request.
+		config, err = client.UpdateConfigInheritable(ctx, project, name, inheritable)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	descriptors, nil := inheritsArgToDescriptors(inherits)
```

Comment:
I fixed this in my patch but assignment to `nil` probably isn't what you intended here